neqo-http3/src/connection_client.rs

```diff
@@ -5930,4 +5938,113 @@ mod tests {
         };
         assert!(server.conn.events().any(stop_sending_event));
     }
+
+    fn maybe_authenticate(conn: &mut Http3Client) {
+        let authentication_needed = |e| matches!(e, Http3ClientEvent::AuthenticationNeeded);
+        if conn.events().any(authentication_needed) {
+            conn.authenticated(AuthenticationStatus::Ok, now());
+        }
+    }
+
+    // Test that decoder stream type is always sent beofer any other instruction also
+    // in case when 0RTT is used.
+    #[test]
+    fn zerortt_request_use_dynamic_table() {
+        const MAX_TABLE_SIZE: u64 = 65536;
+        const MAX_BLOCKED_STREAMS: u16 = 5;
+        let mut client = default_http3_client_param(MAX_TABLE_SIZE);
+        let mut server = Http3Server::new(
+            now(),
+            DEFAULT_KEYS,
+            DEFAULT_ALPN_H3,
+            anti_replay(),
+            Rc::new(RefCell::new(FixedConnectionIdManager::new(5))),
+            QpackSettings {
+                max_table_size_encoder: MAX_TABLE_SIZE,
+                max_table_size_decoder: MAX_TABLE_SIZE,
+                max_blocked_streams: MAX_BLOCKED_STREAMS,
+            },
+        )
+        .unwrap();
+
+        let mut datagram = None;
+        let is_done = |c: &Http3Client| matches!(c.state(), Http3State::Connected);
+        while !is_done(&mut client) {
+            maybe_authenticate(&mut client);
+            datagram = client.process(datagram, now()).dgram();
+            datagram = server.process(datagram, now()).dgram();
+        }
+
+        // exchannge qpack settings, server will send a token as well.
+        datagram = client.process(datagram, now()).dgram();
+        datagram = server.process(datagram, now()).dgram();
+        let _ = client.process(datagram, now()).dgram();
+
+        let token = client
+            .events()
+            .find_map(|e| {
+                if let Http3ClientEvent::ResumptionToken(token) = e {
+                    Some(token)
+                } else {
+                    None
+                }
+            })
+            .unwrap();
```

Comment:
Everything above here is basically setup for 0-RTT.  Make a function that runs client and server and spits out a token.  (It might have to take the server as an argument, but the client can be constructed in that function.